### PR TITLE
snippet: skip true items in UltiSnips list

### PR DESCRIPTION
**Pull request by sl8vz**
Signed-off-by: Seb Laveze <sl8vz@users.noreply.github.com>

### CODE REVIEW

**Comment by sl8vz** [comment]
UlltiSnips always return something even if there is no match. So I was always getting a "True" item, preventing notably to automatically switch to other completion sources automatically. The element is key = true, val = 6.

**Comment by stevearc** [comment]
I've found that this is actually a problem with `api.nvim_call_function`. Calling `api.nvim_call_function` on a vimscript function that simply returns `{}` produces the same result. If you change
```lua
local snippetsList = api.nvim_call_function('UltiSnips#SnippetsInCurrentScope', {})
```
to
```lua
local snippetsList = vim.call('UltiSnips#SnippetsInCurrentScope')
```
it fixes the problem. Perhaps there's a bug with nvim_call_function, as I haven't seen any documentation about serialization differences.

**Comment by sl8vz** [comment]
Indeed @stevearc that's the way to go. You open a PR and I close this one ? Or I can make the change as well, no problem.

**Comment by stevearc** [comment]
I think it's easiest to just change it here; I didn't want to open a second PR for the same issue.

I'll see about reporting the nvim_call_function behavior to neovim

File: lua/completion/source/snippet.lua [hunk 1]
```diff
@@ -16,10 +16,6 @@ M.getUltisnipItems = function(prefix)
   local kind = 'UltiSnips'
   kind = opt.get_option('customize_lsp_label')[kind] or kind
   for key, val in pairs(snippetsList) do
-    -- fix lua parsing issue
-    if key == true then
-      key = 'true'
-    end
```
Comment (sl8vz) [inline lua/completion/source/snippet.lua:22]: I believe this check was due to issues with nvim_call_function

**Comment by sl8vz** [comment]
Note: I haven't changed: https://github.com/nvim-lua/completion-nvim/blob/936bbd17577101a4ffb07ea7f860f77dd8007d43/lua/completion/source/snippet.lua#L63
since the handling is a bit different I don't want to break something I can't test.

**Comment by stevearc** [comment]
Sean had a great explanation on the linked neovim issue https://github.com/neovim/neovim/issues/13524
It's not a bug, but a feature of how the msgpack serialization works. `vim.call` doesn't exhibit this quirk, and seems to be the correct approach.

**Comment by haorenW1025** [comment]
Thanks for the fix! This is super annoying before:)

**Pull request merged**